libevm/VM.cpp

```diff
@@ -815,7 +815,93 @@ void VM::interpretCases()
 			m_sp -= 2;
 		CASE_END
 
+		CASE_BEGIN(JUMPTO)
+		THROW_IF_SUBS_DISABLED();
+			ON_OP();
+			updateIOGas();
+			{
+				uint64_t dest       = m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				m_pc =    dest;
+			}
+		CASE_END
+
+		CASE_BEGIN(JUMPIF)
+		THROW_IF_SUBS_DISABLED();
+			ON_OP();
+			updateIOGas();
+			if (*m_sp--) {
+				uint64_t dest       = m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				m_pc =   dest;
+			} else
+				++m_pc;
+			--m_sp;
+		CASE_END
+
 		CASE_BEGIN(JUMPV)
+		THROW_IF_SUBS_DISABLED();
+			ON_OP();
+			updateIOGas();			
+			{
+				byte n = *(byte*)m_pc;
+				uint64_t i = uint64_t(*m_sp--);
+				if (i >= n) i = n - 1;
+				i = 1 + i * 4;
+				uint64_t dest       = m_code[i++];
+				dest =  (dest << 8) | m_code[i++];
+				dest =  (dest << 8) | m_code[i++];
+				dest =  (dest << 8) | m_code[i];
+				m_pc =   dest;
+			}
+		CASE_END
+
+		CASE_BEGIN(JUMPSUB)
+		THROW_IF_SUBS_DISABLED();
+			ON_OP();
+			updateIOGas();
+			{
+				*++m_rp = m_pc;
+				uint64_t dest       = m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc++];
+				dest =  (dest << 8) | m_code[m_pc];
+				m_pc =   dest;
+			}
+		CASE_END
+
+		CASE_BEGIN(JUMPSUBV)
+		THROW_IF_SUBS_DISABLED();
+			ON_OP();
+			updateIOGas();
+			{
+				*++m_rp = m_pc;
+				byte n = *(byte*)m_pc;
+				uint64_t i = uint64_t(*m_sp--);
+				if (i >= n) i = n - 1;
+				i = 1 + i * 4;
```

Comment:
Please either add some comments about how jumpsubv works or use variable names that are not single letters :-)